src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ModelName/OptionsSuffixAnalyzer.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Azure.ClientSdk.Analyzers.ModelName
+{
+    /// <summary>
+    /// Analyzer that checks model names ending with "Options". This analyzer shares the diagnostic ID with <see cref="GeneralSuffixAnalyzer"/>.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class OptionsSuffixAnalyzer : SuffixAnalyzerBase
+    {
+        private const string AzureResourceManagerNamespaceName = "Azure.ResourceManager";
+        private const string OptionsSuffix = "Options";
+        private const string IUtf8JsonSerializable = "IUtf8JsonSerializable";
+        private const string JsonElement = "JsonElement";
+        private static readonly string[] suffixes = new string[] { OptionsSuffix };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Descriptors.AZC0030); } }
```

Comment:
This pattern will allocate each time the property is called.   Since the underlying array is immutable, we should be able to share the same instance.
```suggestion
        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } => 
            ImmutableArray.Create(Descriptors.AZC0030);
```